app/lib/common/widgets/share/widgets/external_share_options.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:flutter/material.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+
+class ExternalShareOptions extends StatelessWidget {
+  final String? sectionTitle;
+  final GestureTapCallback? onTapCopy;
+  final GestureTapCallback? onTapQr;
+  final GestureTapCallback? onTapSignal;
+  final GestureTapCallback? onTapWhatsApp;
+  final GestureTapCallback? onTapTelegram;
+  final GestureTapCallback? onTapMore;
+
+  const ExternalShareOptions({
+    super.key,
+    this.sectionTitle,
+    this.onTapCopy,
+    this.onTapQr,
+    this.onTapSignal,
+    this.onTapWhatsApp,
+    this.onTapTelegram,
+    this.onTapMore,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final lang = L10n.of(context);
+
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      mainAxisAlignment: MainAxisAlignment.start,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        if (sectionTitle != null) ...[
+          Row(
+            children: [
+              Divider(indent: 0),
+              Text(
+                sectionTitle!,
+                textAlign: TextAlign.start,
+                style: Theme.of(context).textTheme.labelLarge,
+              ),
+              Expanded(child: Divider(indent: 20)),
+            ],
+          ),
+          SizedBox(height: 12),
+        ],
+        SingleChildScrollView(
+          scrollDirection: Axis.horizontal,
+          child: Row(
+            children: [
+              if (onTapQr != null)
+                shareToItemUI(
+                  name: lang.qr,
+                  iconData: PhosphorIcons.qrCode(),
+                  color: Colors.grey.shade600,
+                  onTap: onTapQr,
+                ),
+              if (onTapCopy != null)
+                shareToItemUI(
+                  name: lang.copyLink,
+                  iconData: PhosphorIcons.link(),
+                  color: Colors.blueGrey,
+                  onTap: onTapCopy,
+                ),
+              if (onTapSignal != null)
+                shareToItemUI(
+                  name: lang.signal,
+                  iconData: PhosphorIcons.chat(),
+                  color: Colors.deepPurpleAccent,
+                  onTap: onTapSignal,
+                ),
+              if (onTapWhatsApp != null)
+                shareToItemUI(
```

Comment:
we should also check that whatsApp is available on the system in general, no? same for all the others, too ...

Reply:
Yes, this is in my TO-DO List. Planing to add it next upcoming PR.